src/Microsoft.ServiceFabric.Services.Remoting/Description/InterfaceDescription.cs

```diff
@@ -67,10 +67,13 @@ public MethodDescription[] Methods
             MethodReturnCheck methodReturnCheck)
         {
             EnsureValidMethods(remotedInterfaceKindName, remotedInterfaceType, methodReturnCheck);
-            var methods = remotedInterfaceType.GetMethods().Select(
-                methodInfo => MethodDescription.Create(remotedInterfaceKindName, methodInfo)).ToList();
-
-            return methods.ToArray();
+            var methods = remotedInterfaceType.GetMethods();
```

Comment:
Absolutely. In the existing code, we are:
1) getting the array of methods
2) projecting the array to an IEnumerable, with .Select(...)
3) creating a List (with resizes from 0->4->8->16->etc. on the internal array because ToList only honors the Count when it is an ICollection which in this case, it is not)
4) creating an array from the list (another array and copy operation)

Since we know the length of the array, we can make a single array of the projected type and avoid all extraneous resize and copy operations.

I can submit the code to look at performance this evening.